homeassistant/components/husqvarna_automower/__init__.py

```diff
@@ -0,0 +1,65 @@
+"""The Husqvarna Automower integration."""
+
+import contextlib
+import logging
+
+from aiohttp import ClientError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import aiohttp_client, config_entry_oauth2_flow
+
+from . import api
+from .const import DOMAIN
+from .coordinator import AutomowerDataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+PLATFORMS: list[Platform] = [
+    Platform.LAWN_MOWER,
+]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up this integration using UI."""
+    hass.data.setdefault(DOMAIN, {})
+    implementation = (
+        await config_entry_oauth2_flow.async_get_config_entry_implementation(
+            hass, entry
+        )
+    )
+    session = config_entry_oauth2_flow.OAuth2Session(hass, entry, implementation)
+    hass.data[DOMAIN][entry.entry_id] = api.AsyncConfigEntryAuth(
+        aiohttp_client.async_get_clientsession(hass), session
+    )
+    try:
+        await session.async_ensure_token_valid()
+    except ClientError as err:
+        raise ConfigEntryNotReady from err
+
+    coordinator = AutomowerDataUpdateCoordinator(
+        hass,
+        implementation,
+        session,
+    )
+    await coordinator.async_config_entry_first_refresh()
+
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = coordinator
+
+    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Handle unload of an entry."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+    with contextlib.suppress(Exception):
+        await coordinator.session.close()
```

Comment:
We also need to clean up running tasks on Home Assistant stop event. Search for `EVENT_HOMEASSISTANT_STOP` for examples.